src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1688,7 +1688,7 @@ private bool IsReadOnlyCall(MethodSymbol method, NamedTypeSymbol methodContainin
         {
             Debug.Assert(methodContainingType.IsVerifierValue(), "only struct calls can be readonly");
 
-            if (methodContainingType.IsReadOnly && method.MethodKind != MethodKind.Constructor)
+            if ((method.IsReadOnly || methodContainingType.IsReadOnly) && method.MethodKind != MethodKind.Constructor)
```

Comment:
Rather than repeat this double check, method and type, why not add a property `MethodSymbol.IsEffectivelyReadOnly`?

Reply:
I would like to replace `MethodSymbol.IsReadOnly` with:

- `MethodSymbol.IsDeclaredReadOnly`
- `MethodSymbol.IsEffectivelyReadOnly`

This will hopefully avoid the `GetSymbolInfo` versus `GetDeclaredSymbol` kind of confusion.